crypto/evp/p_lib.c

```diff
@@ -328,12 +334,29 @@ EVP_PKEY *EVP_PKEY_new_CMAC_key(ENGINE *e, const unsigned char *priv,
         goto err;
     }
 
+#if 0
     if (EVP_MAC_ctrl(cmctx, EVP_MAC_CTRL_SET_ENGINE, e) <= 0
         || EVP_MAC_ctrl(cmctx, EVP_MAC_CTRL_SET_CIPHER, cipher) <= 0
         || EVP_MAC_ctrl(cmctx, EVP_MAC_CTRL_SET_KEY, priv, len) <= 0) {
         EVPerr(EVP_F_EVP_PKEY_NEW_CMAC_KEY, EVP_R_KEY_SETUP_FAILED);
         goto err;
     }
+#endif
+    /* How to support engines ? */
```

Comment:
I think if the engine param is set then this has to be treated entirely separately. That is essentially an instruction to lookup the provided cipher in an engine provider (i.e. the EVP_MAC_fetch call does something like "engine=yes enginename=blah")

Reply:
Hmmm....although on second thoughts its the cipher that needs to be looked up so it actually needs to be an EVP_CIPHER_fetch call somewhere - but perhaps that should be in the default provider. So maybe this approach of sending the ending name as a param is the right approach after all.